ClientLibrary/Quickstarts/netcore/DualGraphGroupMembershipSample/Program.cs

```diff
@@ -0,0 +1,263 @@
+                foreach (var membership in memberships)
+                {
+                    lookupKeys.Add(new GraphSubjectLookupKey(membership.MemberDescriptor));
+                }
+                memberships.Clear();
+                IReadOnlyDictionary<SubjectDescriptor, GraphSubject> memberDict = graphClient.LookupSubjectsAsync(new GraphSubjectLookup(lookupKeys)).Result;
+                foreach (GraphSubject member in memberDict.Values)
+                {
+                    // member is a user
+                    //switch (member.Descriptor.SubjectType)
+                    //{
+                    //    case "user":
+                    //        users.Add(((GraphUser)member).PrincipalName);
+                    //        break;
+                    //    case 
+
+                    //}
+                    if (member.SubjectKind == "user")
+                    {
+                        users.Add(((GraphUser) member).PrincipalName);
+                    }
+                    // member is an AAD group
+                    else if (member.SubjectKind == "group" && member.Origin == "aad")
+                    {
+                        aadGroups.Add(member.OriginId);
+                    }
+                    // member is a VSTS group
+                    else if (member.SubjectKind == "group" && member.Origin == "vsts")
+                    {
+                        memberships.AddRange(graphClient.GetMembershipsAsync(member.Descriptor,Microsoft.VisualStudio.Services.Graph.GraphTraversalDirection.Down).Result);
+                    }
+                    // should never get here
+                    else
+                    {
+                        throw new Exception("shouldn't be here");
+                    }
+                }
+            }
+            return new Tuple<List<string>, List<string>>(users, aadGroups);
+        }
+
+        private static AuthenticationContext GetAuthenticationContext(string tenant)
+        {
+            AuthenticationContext ctx = null;
+            if (tenant != null)
+                ctx = new AuthenticationContext("https://login.windows-ppe.net/" + tenant);//"https://login.microsoftonline.com/" + tenant);
+            else
+            {
+                ctx = new AuthenticationContext("https://login.windows-ppe.net/common");//"https://login.windows.net/common");
+                if (ctx.TokenCache.Count > 0)
+                {
+                    string homeTenant = ctx.TokenCache.ReadItems().First().TenantId;
+                    ctx = new AuthenticationContext("https://login.windows-ppe.net/" + homeTenant);//"https://login.microsoftonline.com/" + homeTenant);
+                }
+            }
+
+            return ctx;
+        }
+
+        //================ Microsoft graph api helper code ===========================================================
+
+        private static List<string> GetNestedAadGroupUsers(string accessToken, List<string> aadGroupIds)
```

Comment:
The function names are getting confusing. I would preface them by VSTS or AAD to make it clear